topdown/query.go

```diff
@@ -340,6 +341,14 @@ func (q *Query) PartialRun(ctx context.Context) (partials []ast.Body, support []
 	defer q.metrics.Timer(metrics.RegoPartialEval).Stop()
 
 	livevars := ast.NewVarSet()
+	for _, t := range q.unknowns {
+		switch v := t.Value.(type) {
+		case ast.Var:
+			livevars.Add(v)
+		case ast.Ref:
+			livevars.Add(v[0].Value.(ast.Var))
+		}
+	}
```

Comment:
Before, we hadn't added `input` or `data` if they're part of the unknowns to the livevars. This felt a bit odd to me -- what if `input.foo` is unknown, but `input.bar == input.bar` is in the query to-be-copy-propagated? Turns out that's not possible -- we're only running CP after a successful PE run, as a final step. So, if we've made it to CP, and there's a tautology to check (`input.x == input.x`), checking its first term is good enough.